File: CodeChef/ARRGAME.py
t = int(input())
for tt in range(t):
    n = int(input())
    arr = list(map(int,input().split()))
    max_zero = 0
    dct = dict()
    i = 1
    while i <= n-1:
        tmp = 0
        while arr[i] == 0:
            tmp += 1
            i += 1
        if tmp == 0:
            i += 1
            continue
        dct[tmp] = dct.get(tmp,0) + 1
        if tmp > max_zero:
            max_zero = tmp
        i += 1

    x = sorted(dct.keys())
    if len(x) >= 2:
        if max_zero % 2 == 1 and dct[max_zero] == 1 and x[-2] < (max_zero + 1) // 2:
            print("Yes")
        else:
            print("No")
    elif max_zero % 2 == 1 and dct[max_zero] == 1:
        print("Yes")
    else:
        print("No")

File: CodeChef/ARRGAME.java
import java.util.Scanner;

public class ARRGAME {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int t = s.nextInt();
        while(t -- > 0){
            int n = s.nextInt();
            int arr[] = new int[n];
            int max_zeros = 0;
            for(int i =0 ; i<n ; i++){
                arr[i] = s.nextInt();
            }
            for(int i=1 ; i<n-1 ; i++){
                int tmp = 0;
                while(arr[i] == 0){
                    i+=1;
                    tmp += 1;
                }
                if(tmp > max_zeros)
                    max_zeros = tmp;
            }
            if(max_zeros % 2 == 1 )
                System.out.println("Yes");
            else
                System.out.println("No");
        }
        s.close();
    }
}